test/tools/ossfuzz/StackReuseCodegenFuzzer.cpp

```diff
@@ -0,0 +1,153 @@
+    (at your option) any later version.
+
+    solidity is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/tools/ossfuzz/yulProto.pb.h>
+#include <test/tools/ossfuzz/protoToYul.h>
+
+#include <test/tools/ossfuzz/SolidityEvmoneInterface.h>
+
+#include <libyul/AssemblyStack.h>
+#include <libyul/Exceptions.h>
+
+#include <libyul/backends/evm/EVMDialect.h>
+
+#include <libevmasm/Instruction.h>
+
+#include <liblangutil/EVMVersion.h>
+
+#include <src/libfuzzer/libfuzzer_macro.h>
+
+#include <fstream>
+
+using namespace solidity;
+using namespace solidity::test;
+using namespace solidity::test::fuzzer;
+using namespace solidity::yul;
+using namespace solidity::yul::test;
+using namespace solidity::yul::test::yul_fuzzer;
+using namespace solidity::langutil;
+using namespace std;
+
+static evmc::VM evmone = evmc::VM{evmc_create_evmone()};
+
+DEFINE_PROTO_FUZZER(Program const& _input)
+{
+	ProtoConverter converter;
+	string yul_source = converter.programToString(_input);
+	// Fuzzer also fuzzes the EVM version field.
+	langutil::EVMVersion version = converter.version();
+	EVMHost hostContext(version, evmone);
+
+	if (const char* dump_path = getenv("PROTO_FUZZER_DUMP_PATH"))
+	{
+		ofstream of(dump_path);
+		of.write(yul_source.data(), static_cast<streamsize>(yul_source.size()));
+	}
+
+	if (yul_source.size() > 1200)
+		return;
+
+	YulStringRepository::reset();
+
+	auto assemble = [](
+		langutil::EVMVersion _version,
+		solidity::frontend::OptimiserSettings _settings,
+		string const& _yulSource
+	) -> bytes
+	{
+		AssemblyStack stack(
+			_version,
+			AssemblyStack::Language::StrictAssembly,
+			_settings
+		);
+
+		// Parse protobuf mutated YUL code
+		if (
+			!stack.parseAndAnalyze("source", _yulSource) ||
+			!stack.parserResult()->code ||
+			!stack.parserResult()->analysisInfo ||
+			!Error::containsOnlyWarnings(stack.errors())
+		)
+			yulAssert(false, "Proto fuzzer generated malformed program");
+
+		return stack.assemble(AssemblyStack::Machine::EVM).bytecode->bytecode;
+	};
+
+	solidity::frontend::OptimiserSettings settings = solidity::frontend::OptimiserSettings::full();
+	settings.runYulOptimiser = false;
+	settings.optimizeStackAllocation = false;
+	bytes unoptimisedByteCode = assemble(version, settings, yul_source);
+
+	auto initDeployMsg = [](bytes const& _input, evmc_call_kind _callKind) -> evmc_message
+	{
+		// Zero initialize all message fields
+		evmc_message msg = {};
+		// Gas available (value of type int64_t) is set to its maximum
+		// value.
+		msg.gas = std::numeric_limits<int64_t>::max();
+		msg.input_data = _input.data();
+		msg.input_size = _input.size();
```

Comment:
```suggestion
		// assert _input.size <= 0xFFFF
		bytes deployCode = bytes{
			0x38, 0x60, 0x0c, 0x60, 0x00, 0x39, 0x61,
			_input.size() >> 8, _input.size() & 0xFF,
			0x60, 0x00, 0xf3
		} + _input;
		msg.input_data = deployCode.data();
		msg.input_size = deployCode.size();
```

Reply:
Ah, damn, no... ``deployCode`` would need to live outside for the pointer not to be invalidated when returning from the lambda...

Reply:
Maybe just move ``return hostContext.call(msg);`` into the lambda as well, then it should be fine.

Reply:
Shouldn't the first patch be the following instead (essentially `0xc` instead of `0x0` just before the return instruction i.e., `0xf3`?

```
		bytes deployCode = bytes{
			0x38, 0x60, 0x0c, 0x60, 0x00, 0x39, 0x61,
			_input.size() >> 8, _input.size() & 0xFF,
			0x60, 0x0c, 0xf3
		} + _input;
```

Reply:
Please ignore my comment.